doc/imageview_warmups.md

```diff
@@ -0,0 +1,329 @@
+        update = Node(true)
+        c = new(io, s1, s2, update)
+        throttled = throttle(1/60, update)
+        # The node that gets returned by `map` will be garbage-collected
+        # unless we call `preserve` on it. An alternative is to store
+        # this node somewhere (e.g., see the "TwoSharedNodes" example below).
+        Reactive.preserve(map(x->println(c.io, "state1: $(c.state1); state2: $(c.state2)"), throttled))
+        c
+    end
+end
+
+Canvas(io::IO, s1, s2) = Canvas{typeof(io)}(io, s1, s2)
+
+state1!(c::Canvas, val) = (c.state1 = val; push!(c.update, true); val)
+state2!(c::Canvas, val) = (c.state2 = val; push!(c.update, true); val)
+
+end  # module
+
+using Reactive, TwoStates
+
+eventqueue = Timer(_->Reactive_run_till_now(), 0.001, 0.001)
+
+# OK, let's try it!
+c = Canvas(STDOUT, 1, 1)
+state1!(c, 5)
+sleep(0.1)
+
+state1!(c, 7)
+state2!(c, -3)
+```
+
+Here's the output:
+```jl
+julia> include("twostates.jl")
+state1: 1; state2: 1
+state1: 5; state2: 1
+-3
+
+julia> state1: 7; state2: -3
+julia>
+```
+Note that your exact output can depend on the details of timing between
+`eventloop` and the REPL code.
+
+The first output line was triggered by creating the `Canvas`. The
+second line was triggered by the first `state1!` call; most likely,
+both of these outputs were produced during the `sleep`. The next line
+is the returned value from `state2!(c, -3)`, and control returns to
+the julia REPL. However, roughly a millisecond later the event queue
+fires and processes updates; it produces a single line of output for
+the updates to both states. Without the `throttle`, we instead would
+have gotten something like
+
+```jl
+julia> include("twostates.jl")
+state1: 1; state2: 1
+state1: 5; state2: 1
+state1: 7; state2: -3
+-3state1: 7; state2: -3
+
+julia>
+```
+
+Note that unlike the previous case using `throttle`, here each
+`state!` call generated a corresponding line of output.
+
+
+Interestingly, the output produced by the second update of `signal1`
+also included the consequences of updating `signal2`: this is because
+the updates happened before the event loop fired again, so by the time
+the `println` statement ran both values had already been updated.
+In some circumstances (like this one), this behavior might be fine or
+even desirable; in other cases, such behavior could be a source of
+bugs.
+
+So in the spirit of exploration, let's look at a second implementation
+that preserves history:
+
+```jl
+module TwoNodes
+
+using Reactive
+
+export Canvas, state1!, state2!
+
+type Canvas{S<:IO}
+    io::S
+    state1::Node{Int}
+    state2::Node{Int}
+
+    function Canvas(io::S, s1::Integer, s2::Integer)
+        n1, n2 = Node(Int, s1), Node(Int, s2)
+        c = new(io, n1, n2)
+        combined = merge(n1, n2)
+        throttled = throttle(1/60, combined)
+        Reactive.preserve(map(x->println(c.io, "state1: $(value(c.state1)); state2: $(value(c.state2))"), throttled))
```

Comment:
I find the use of the `value` slightly awkward.... But it's really OK. Here's an alternative

`show_canvas(c) = map(render, sampleon(throttle(1/60, merge(c.state1, state2)), Node(c)))`

(I find it better to take out the rendering from the type definition as a personal preference)